src/osx/setup-env.cpp

```diff
@@ -9,17 +9,41 @@
 
 #include "filesystem.h"
 
+namespace {
+void setenvPathIfExists(const char* name, fs::path const& path, bool override) {
+    if (fs::exists(path)) {
+        setenv(name, path.c_str(), 1);
+    }
+}
+
+}  // namespace
 
 void setupEnvironment() {
     /**
      * see https://gitlab.gnome.org/GNOME/gtk-mac-bundler/-/blob/master/examples/gtk3-launcher.sh
      * and https://gitlab.gnome.org/GNOME/gtk-mac-bundler/-/issues/12
      */
+
+    // Get the folder containing the application executable
     auto base = Stacktrace::getExePath().parent_path();  // Xournal++.app/Contents or $HOME/gtk/inst
 
-    if (fs::exists(base / "Resources")) {  // app-bundle
+    /* We consider the application to launch from a bundle iff the parent
+     * directory is named "Content" and it has a "Resources" folder.
+     * Note that checking the latter only is insufficient, e.g. it will
+     * falsely detect a bundle from the build folder, since the repository root
+     * has a folder named "resources"
+     */
+    const bool isInBundle = base.filename() == "Content" && fs::exists(base / "Resources");
```

Comment:
```suggestion
     * directory is named "Contents" and it has a "Resources" folder.
     * Note that checking the latter only is insufficient, e.g. it will
     * falsely detect a bundle from the build folder, since the repository root
     * has a folder named "resources"
     */
    const bool isInBundle = base.filename() == "Contents" && fs::exists(base / "Resources");
```